test/e2e/framework/util.go

```diff
@@ -643,7 +689,7 @@ func WaitForPodsRunningReady(c *client.Client, ns string, minPods int32, timeout
 		logPodStates(badPods)
 		return false, nil
 	}) != nil {
-		return fmt.Errorf("Not all pods in namespace '%s' running and ready within %v", ns, timeout)
+		return fmt.Errorf(errorBadPodsStates(badPods, nPods, ns, timeout))
```

Comment:
1. If you want to do something of this form, it's safer to do `fmt.Errorf("%s", functionCall())`. Otherwise `%` characters will mess up your output.
2. Don't do this at all, there's an errors.New() function that takes a string.
3. Just make the func return an error in the first place.